s3o/s3o.go

```diff
@@ -74,16 +75,41 @@ func fetchPubkey() (*rsa.PublicKey, error) {
 // Handler wraps the given handler in s3o authentication
 func Handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
 		if err := r.ParseForm(); err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
-
 		user := r.Form.Get("username")
 		token := r.Form.Get("token")
 
-		if user == "" || token == "" {
+		// These parameters come from https://s3o.ft.com. It redirects back after it does the google authentication.
+		if r.Method == "POST" && user != "" {
+			code, err := authenticateToken(user, token, r.Host)
+			if err != nil {
+				w.WriteHeader(code)
+				fmt.Fprint(w, err.Error())
+			}
+
+			// set cookies
+			http.SetCookie(w, &http.Cookie{Name: "s3o_username", Value: user, MaxAge: 900000, HttpOnly: true})
```

Comment:
Maybe worth extracting the cookie names into constants and use them here and in the `isAuthFromCookie` method